File: solutions/fsharp/collatz-conjecture/1/CollatzConjecture.fs
module CollatzConjecture

let steps (number: int): int option = 
    let rec collatzHelper counter number =
        if number = 1 then
            Some counter
        elif number % 2 = 0 then
            collatzHelper (counter+1) (number / 2)
        else
            collatzHelper (counter+1) (3*number+1)
    if number <= 0 then
        None
    else
        collatzHelper 0 number

File: solutions/python/collatz-conjecture/1/collatz_conjecture.py
def steps(number: int) -> int:
    if number < 1:
        raise ValueError("Only positive integers are allowed")

    counter = 0

    while number != 1:
        counter += 1

        if number % 2 == 0:
            number //= 2
            continue

        number = 3 * number + 1

    return counter
